trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js

```diff
@@ -30,54 +30,60 @@ const updatedIssuedFacilities = async (deal) => {
 
     if (shouldUpdateFacility) {
+      let facilityPremiumSchedule;
+      let feeRecord;
+      let facilityUpdate;
+
       // update portal facility status
       const facilityStatusUpdate = CONSTANTS.FACILITIES.FACILITY_STATUS_PORTAL.ACKNOWLEDGED;
-      await api.updatePortalFacilityStatus(facilityId, facilityStatusUpdate);
-
-      // add acknowledged flag to Portal facility
-      const portalFacilityUpdate = {
-        hasBeenAcknowledged: true,
-      };
 
-      const updatedPortalFacility = await api.updatePortalFacility(facilityId, portalFacilityUpdate);
+      // Add `hasBeenAcknowledged` flag to non-GEF facility
+      if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS) {
+        await api.updatePortalFacilityStatus(facilityId, facilityStatusUpdate);
+
+        const portalFacilityUpdate = {
+          hasBeenAcknowledged: true,
+        };
+
+        const updatedPortalFacility = await api.updatePortalFacility(facilityId, portalFacilityUpdate);
+        facility.hasBeenAcknowledged = updatedPortalFacility.hasBeenAcknowledged;
+        facility.status = facilityStatusUpdate;
+        facilityUpdate = {
+          ...portalFacilityUpdate,
+        };
+      }
 
       // update TFM facility
       const facilityExposurePeriod = await getFacilityExposurePeriod(facility);
-
       const facilityGuaranteeDates = getGuaranteeDates(facility, dealSubmissionDate);
 
-      let facilityPremiumSchedule;
+      // Premium Schedule is only valid for non-GEF facilities
       if (dealType === CONSTANTS.DEALS.DEAL_TYPE.BSS_EWCS) {
         facilityPremiumSchedule = await getFacilityPremiumSchedule(
           facility,
           facilityExposurePeriod,
           facilityGuaranteeDates,
         );
+        facilityUpdate = {
+          premiumSchedule: facilityPremiumSchedule,
+        };
+      } else if (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF) {
+      // Fee record is only valid for GEF facilities
+        if (submissionType !== CONSTANTS.DEALS.SUBMISSION_TYPE.MIA) {
+          feeRecord = calculateGefFacilityFeeRecord(facility);
+          facilityUpdate = {
+            feeRecord,
+          };
+        }
       }
 
-      let feeRecord;
-
-      const shouldCalculateFeeRecord = (dealType === CONSTANTS.DEALS.DEAL_TYPE.GEF
-        && submissionType !== CONSTANTS.DEALS.SUBMISSION_TYPE.MIA);
-
-      if (shouldCalculateFeeRecord) {
-        feeRecord = calculateGefFacilityFeeRecord(facility);
-      }
-
-      const facilityUpdate = {
-        ...portalFacilityUpdate,
+      facilityUpdate = {
+        ...facilityUpdate,
         ...facilityExposurePeriod,
         facilityGuaranteeDates,
-        premiumSchedule: facilityPremiumSchedule,
-        feeRecord,
       };
 
       const updateFacilityResponse = await api.updateFacility(facilityId, facilityUpdate);
-
-      // add the updated properties to the returned facility
-      // to retain flat, generic facility mapping used in deal submission calls.
```

Comment:
I think we should keep this comment in there (but in the new place), incase someone wonders why we do this